agent_plugin_builder/build_plugin.py

```diff
@@ -0,0 +1,252 @@
+    """
+    import shutil
+
+    if not plugin_path.exists():
+        logger.error(f"Plugin path {plugin_path} does not exist")
+        raise FileNotFoundError(f"Plugin path {plugin_path} does not exist")
+
+    if not build_dir_path.exists():
+        logger.info(f"Creating build directory: {build_dir_path}")
+        build_dir_path.mkdir(exist_ok=True)
+    else:
+        try:
+            logger.info(f"Clearing build directory: {build_dir_path}")
+            shutil.rmtree(build_dir_path)
+        except shutil.Error as err:
+            logger.error(f"Unable to clear build directory: {build_dir_path}")
+            raise err
+
+    try:
+        logger.info(f"Copying plugin code to build directory: {plugin_path} -> {build_dir_path}")
+        shutil.copytree(plugin_path, build_dir_path, dirs_exist_ok=True)
+    except shutil.Error as err:
+        logger.error(
+            f"Unable to copy plugin code to build directory: {plugin_path} -> {build_dir_path}"
+        )
+        raise err
+
+    agent_plugin_manifest = get_agent_plugin_manifest(build_dir_path)
+    create_agent_plugin_archive(build_dir_path, agent_plugin_manifest, dist_dir_path)
+
+
+def get_agent_plugin_manifest(build_dir_path: Path) -> AgentPluginManifest:
+    """
+    Get the Agent Plugin manifest from the build directory.
+
+    :param build_dir_path: Path to the build directory.
+    """
+    manifest_file_path = _get_plugin_manifest_filename(build_dir_path)
+
+    logger.info(f"Reading plugin manifest file: {manifest_file_path}")
+    with manifest_file_path.open("r") as f:
+        return AgentPluginManifest(**yaml.safe_load(f))
+
+
+def create_agent_plugin_archive(
+    build_dir_path: Path,
+    agent_plugin_manifest: AgentPluginManifest,
+    dist_dir_path: Path,
+):
+    """
+    Create the Agent Plugin tar archive.
+
+    :param build_dir_path: Path to the build directory.
+    :param agent_plugin_manifest: Agent Plugin manifest.
+    :param dist_dir_path: Path to the dist directory.
+    """
+    build_options = parse_agent_plugin_build_options(build_dir_path)
+    dependency_method = build_options.platform_dependencies
+    generate_vendor_directories(build_dir_path, agent_plugin_manifest, dependency_method)
+    generate_plugin_config_schema(build_dir_path, agent_plugin_manifest)
+    create_source_archive(build_dir_path)
+    create_plugin_archive(build_dir_path, agent_plugin_manifest, dist_dir_path)
+
+
+def generate_vendor_directories(
+    build_dir_path: Path,
+    agent_plugin_manifest: AgentPluginManifest,
+    dependency_method: PlatformDependencyPackagingMethod,
+):
+    """
+    Generate the vendor directories for the plugin. If the dependency method is not chosen
+    or if the plugin supports multiple operating systems, the method will try to generate
+    common vendor directories. If common vendor directory are not possible, it will generate
+    separate vendor directories for each supported operating system.
```

Comment:
"If the dependency method is not chosen"

Since this can't be `None`, a method is always chosen.